src/main/scala/fi/oph/koski/raportit/PerusopetuksenRaportitRepository.scala

```diff
@@ -196,8 +202,75 @@ case class PerusopetuksenRaportitRepository(db: DB) extends KoskiDatabaseMethods
       group by
         oo.opiskeluoikeus_oid"""
   }
+
+  // Tarkoituksena on hakea raportille tapauksia jotka ovat potentiaalisesti keskeyttäneet opinnot, mutta tätä tietoa ei ole siirretty Koskeen.
+  private def haettuVuosiluokkaViimeisinVahvistettuSuoritus(oppilaitos: Set[Organisaatio.Oid], paiva: LocalDate, vuosiluokka: String): Seq[Tunnisteet] = {
+    aktiivistenSuoritukset(oppilaitos.toSeq, Date.valueOf(paiva), vuosiluokka)
+      .foldLeft[List[Tunnisteet]](Nil)(viimeisinSuoritusOnHaettuVuosiluokka(vuosiluokka))
+  }
+
+  private def viimeisinSuoritusOnHaettuVuosiluokka(vuosiluokka: String)(acc: List[Tunnisteet], curr: (OpiskeluoikeusOid, Seq[VuosiluokanTiedot], Seq[AikajaksoId])): List[Tunnisteet] = {
+    val (opiskeluoikeusOid, päätasonIdsJaVuosiluokat, aikajaksoIds) = curr
+
+    val päätasonSuoritusIds = päätasonIdsJaVuosiluokat
+      .takeWhile { x => x.vuosiluokka == vuosiluokka && x.vahvistuspäivä.isDefined }
+      .map(_.id)
+
+    päätasonSuoritusIds match {
+      case Nil => acc
+      case ids => (opiskeluoikeusOid, ids, aikajaksoIds) :: acc
+    }
+  }
+
+  private def aktiivistenSuoritukset(oppilaitos: Seq[String], paiva: Date, vuosiluokka: String) = {
+    import fi.oph.koski.db.PostgresDriverWithJsonSupport.plainAPI._
+    implicit val getResult = GetResult(r => (r.nextString(), r.nextArray(), r.nextArray()))
+    val queryResult = runDbSync(aktiivistenSuorituksetQuery(oppilaitos, paiva, vuosiluokka).as[(OpiskeluoikeusOid, Seq[String], Seq[AikajaksoId])], timeout = 5.minutes)
+
+    queryResult.map { case (opiskeluoikeusOid, jsonStrings, aikajaksoIds) =>
+      val parsed = jsonStrings
+        .map(JsonMethods.parse(_))
+        .map(JsonSerializer.extract[VuosiluokanTiedot](_))
+
+      (opiskeluoikeusOid, parsed, aikajaksoIds)
+    }
+  }
+
+  private def aktiivistenSuorituksetQuery(oppilaitos: Seq[String], paiva: Date, vuosiluokka: String) = {
```

Comment:
Mun mielestä ois siistimpää jos tämä metodi tekisi myös `runDbSync`:in. Ei tarvitsisi kahdessa paikassa importtailla plain-apeja jne.
Pitäisikö `oppilaitos` olla `oppilaitokset` jos niitä on monta?